src/modules/ekf_att_pos_estimator/ekf_att_pos_estimator_main.cpp

```diff
@@ -1260,14 +1270,65 @@ void AttitudePositionEstimatorEKF::pollData()
 	//Update AirSpeed
 	orb_check(_airspeed_sub, &_newAdsData);
 
-	if (_newAdsData) {
-		orb_copy(ORB_ID(airspeed), _airspeed_sub, &_airspeed);
-		perf_count(_perf_airspeed);
+	/* For VTOLs reject airspeed measurments in rw mode (hover) */
+	if (_vstatus.is_vtol) {
+		orb_check(_vtol_vstatus_sub, &_newVtolStatus);
+		
+		if (_newVtolStatus) {
+			orb_copy(ORB_ID(vtol_vehicle_status), _vtol_vstatus_sub, &_vtol_vstatus);
+		}
+
+		if (_vtol_vstatus.vtol_in_rw_mode) {
```

Comment:
This is the wrong architectural approach. What you *really* intended was to reset the wind states when transitioning from a high AoA situation into a low AoA situation. That is what you should be conditioning on, and not on the VTOL state. You're building the assumption that you have a tailsitter into the filter here, and you're assuming that you're already transitioning into it as well. Both are not good assumptions to make.